rust/xaynet-client/src/mobile_client/participant/mod.rs

```diff
@@ -32,19 +34,93 @@ pub struct ParticipantState {
     pub keys: SigningKeyPair,
     // Mask config
     pub aggregation_config: AggregationConfig,
+    pub max_message_size: MaxMessageSize,
 }
 
 #[derive(Serialize, Deserialize)]
 pub struct ParticipantSettings {
     pub secret_key: ParticipantSecretKey,
     pub aggregation_config: AggregationConfig,
+    pub max_message_size: MaxMessageSize,
+}
+
+const MESSAGE_HEADER_LEN: usize = 136;
+const ENCRYPTION_HEADER_LEN: usize = 48;
+pub const MIN_MESSAGE_SIZE: usize = MESSAGE_HEADER_LEN + ENCRYPTION_HEADER_LEN + 1;
+#[derive(Debug, Error)]
+#[error("max message size must be at least {}", MIN_MESSAGE_SIZE)]
+pub struct InvalidMaxMessageSize;
+
+/// Represent the maximum size messages sent by a participant can
+/// have. If a larger message needs to be sent, it will be chunked and
+/// sent in several parts. Note that messages have a minimal size of
+/// [`MIN_MESSAGE_SIZE`].
+#[derive(Serialize, Deserialize, Clone, Copy, Debug)]
+pub struct MaxMessageSize(
+    #[serde(deserialize_with = "deserialize_max_message_size")] Option<usize>,
+);
```

Comment:
this represents no splitting with `None` whereas the multipart encoding uses `0`. it would be nice to unify this to use only one of the two approaches, maybe in a later refactoring pr, i guess the `Option<usize>` is preferable over the `usize`.

Reply:
Yeah I agree. I hesitated but wanted to keep the changes small for this PR. The encoder also uses the _payload_ size while the settings use the _encrypted message_ size. We should also consolidate that as well.